aip/0162.md

```diff
@@ -111,6 +111,44 @@ user sent a name with no revision ID, the returned `name` string **must not**
 include the revision ID either. Similarly, if the user sent a name with a
 revision ID, the returned `name` string **must** explicitly include it.
 
+### Tagging revisions
+
+APIs implementing resource revisions **may** provide a mechanism for users to
+tag a specific revision with a user provided name by implementing a `Tag`
+custom method:
+
+```proto
+rpc TagBookRevision(TagBookRevisionRequest) returns (Book) {
+  option (google.api.http) = {
+    post: "/v1/{name=publishers/*/books/*}:tagRevision"
+    body: "*"
+  }
+}
+```
+
+```proto
+message TagBookRevisionRequest {
+  // The name of the book to be tagged, including the revision ID.
+  string name = 1;
+
+  // The tag to apply.
+  // The tag should be at most 40 characters, and match `[a-z][a-z0-9-]{3,39}`.
+  string tag = 2;
+}
+```
+
+- The `name` field **should** require an explicit revision ID to be provided.
+- Once a revision is tagged, the API **must** support using the tag in place of
+  the revision ID in `name` fields.
+  - If the user sends a tag, the API **must** return the tag in the resource's
+    `name` field, but the revision ID in the resource's `revision_id` field.
+  - If the user sends a revision ID, the API **must** return the revision ID in
+    both the `name` field and the `revision_id` field.
+- If the user calls the `Tag` method with an existing tag, the request **must**
+  succeed and the tag updated to point to the new requested revision ID. This
+  allows users to write code against specific tags (e.g. `published`) and the
+  revision can change in the background with no code change.
```

Comment:
Maybe add something about listing tags? I can imagine that being a pretty frequently-used option.

Reply:
Good point.